modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/tcs/Tcs.scala

```diff
@@ -155,12 +155,14 @@ object Tcs {
     }
   }
 
-  def fromSequenceConfig(config: Config)(s0: TcsConfig): TcsConfig =
+  def fromSequenceConfig(config: Config, subsystems: NonEmptyList[Subsystem])(s0: TcsConfig): TcsConfig = {
+    val subs = subsystems.toList
     List(
-      build(buildPwfs1Config,  TELESCOPE_KEY / GUIDE_WITH_PWFS1_PROP, config),
-      build(buildPwfs2Config,  TELESCOPE_KEY / GUIDE_WITH_PWFS2_PROP, config),
-      build(buildOiwfsConfig,  TELESCOPE_KEY / GUIDE_WITH_OIWFS_PROP, config),
-      build(buildOffsetConfig, TELESCOPE_KEY / P_OFFSET_PROP, TELESCOPE_KEY / Q_OFFSET_PROP, config)
-    ).suml.apply(s0)
+      subs.contains(Subsystem.P1WFS).option(build(buildPwfs1Config, TELESCOPE_KEY / GUIDE_WITH_PWFS1_PROP, config)),
+      subs.contains(Subsystem.P2WFS).option(build(buildPwfs2Config, TELESCOPE_KEY / GUIDE_WITH_PWFS2_PROP, config)),
+      subs.contains(Subsystem.OIWFS).option(build(buildOiwfsConfig, TELESCOPE_KEY / GUIDE_WITH_OIWFS_PROP, config)),
+      subs.contains(Subsystem.Mount).option(build(buildOffsetConfig, TELESCOPE_KEY / P_OFFSET_PROP, TELESCOPE_KEY / Q_OFFSET_PROP, config))
+    ).collect{case Some(x) => x}.suml.apply(s0)
```

Comment:
Couldn't you just do `.flatten` here instead of `.collect`?
I think it removes all `None` entries and unpacks the `Some(x)` to `x`.

Reply:
```
scala> List(5.some, None, 3.some, None).flatten
res14: List[Int] = List(5, 3)
```

Reply:
Wart remover doesn't let us use `flatten` for some reason I don't fully understand 😄

Reply:
Wartremover didn't let me.